packages/tiny-agents/src/lib/types.ts

```diff
@@ -21,6 +21,14 @@ export const ServerConfigSchema = z.discriminatedUnion("type", [
 			url: z.union([z.string(), z.string().url()]),
 			options: z
 				.object({
+					/**
+					 * Customizes HTTP requests to the server.
+					 */
+					requestInit: z
+						.object({
+							headers: z.record(z.string()).optional(),
+						})
+						.optional(),
```

Comment:
semi-related comment for @Wauplin, @hanouticelina and @evalstate: 

i am starting to think we should have gone with a simplified configuration schema (the one from VS Code, probably), rather than be more explicit and use the types from the MCP SDK, given here for instance, we need to nest the headers quite a bit inside of the config.

Maybe for a v2 we'll change this! no rush, though

Reply:
well making the switch to something more standard right now is probably better (not in this PR but as follow-up). Agree the `config` nesting is not really needed